src/lib/services/emissionsCalculator.ts

```diff
@@ -36,12 +36,19 @@ export default class EmissionsCalculator {
     energyConsumedWh: number
   ): Promise<number> {
     const url = this.intensityUrl(new Date(new Date(productionStartDate).getTime() - 1000 * 60 * 60), productionEndDate)
-    const response = await fetch(url)
-    if (response.status !== 200) {
-      throw new InternalServerError('Unexpected error fetching carbon intensity data')
+    const hardcodedFactor = 0.07
+    let data: IntensityResponseData | [] = []
+    try {
+      const response = await fetch(url)
+      if (response.status !== 200) {
+        throw new InternalServerError('Unexpected error fetching carbon intensity data')
```

Comment:
Rather than the `throw` can you bring your logic for handling off-line mode into here as well so that we always return a value from this function? Or do we need to keep `InternalServerError`?

```js
logger.info('Unexpected HTTP error fetching carbon intensity data')
return Math.floor(hardcodedFactor * energyConsumedWh)
```

Reply:
If you want i can put a return outside the try {} catch {} block so that it aways returns a value .. ( see my latest commit ) .. 

The throw needs to stay there .. this is in case the carbon API throws 301 but the carbon api still works ..